### PR TITLE
Add missing logging tests

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Part of https://github.com/PyTorchLightning/pytorch-lightning/issues/7898
Follow up to https://github.com/PyTorchLightning/pytorch-lightning/pull/8181/

These tests were removed from a previous PR and we missed adding them back.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [n/a] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

**Review by SkafteNicki — APPROVED** [review]

File: tests/trainer/logging_/test_train_loop_logging.py [hunk 1]
```diff
@@ -690,6 +691,33 @@ def training_step(self, batch, batch_idx):
...
+        def training_step(self, batch, batch_idx):
+            self.log('foo', Accuracy())
+
+    trainer = Trainer(default_root_dir=tmpdir)
+    model = TestModel()
+    with pytest.raises(
+        MisconfigurationException,
+        match=r"`self.log\(foo, ..., metric_attribute=name\)` where `name` is one of \['bar'\]"
+    ):
+        trainer.fit(model)
```
Comment (Borda) [inline tests/trainer/logging_/test_train_loop_logging.py:719]: rather make it as two tests so we run both if anything is failing?

Comment (carmocca) [reply tests/trainer/logging_/test_train_loop_logging.py:719]: The test already includes several other checks. I kind of prefer joining them for these very small checks since it's probably faster for our CI times

**Review by ananthsub — APPROVED** [review]

**Review by Borda — APPROVED** [review]
